pipeline/backend/ssh/ssh.go

```diff
@@ -0,0 +1,141 @@
+package ssh
+
+import (
+	"context"
+	"encoding/base64"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/melbahja/goph"
+
+	"github.com/woodpecker-ci/woodpecker/pipeline/backend/types"
+	"github.com/woodpecker-ci/woodpecker/server"
+)
+
+type ssh struct {
+	cmd    *goph.Cmd
+	output io.ReadCloser
+	client *goph.Client
+}
+
+type readCloser struct {
+	io.Reader
+}
+
+func (c readCloser) Close() error {
+	return nil
+}
+
+// make sure local implements Engine
+var _ types.Engine = &ssh{}
+
+// New returns a new ssh Engine.
+func New() types.Engine {
+	return &ssh{}
+}
+
+func (e *ssh) Name() string {
+	return "ssh"
+}
+
+func (e *ssh) IsAvailable() bool {
+	return true
```

Comment:
```suggestion
	return os.Getenv("WOODPECKER_SSH_KEY_PASSWORD") != "" || os.Getenv("WOODPECKER_SSH_PASSWORD") != ""
```

Something like this could be used.